automated install/basic-install.sh

```diff
@@ -1979,13 +1983,24 @@ FTLcheckUpdate() {
     local localSha1
 
     if [[ ! "${ftlBranch}" == "master" ]]; then
-        #Check whether or not the binary for this FTL branch actually exists. If not, then there is no update!
+        # Check whether or not the binary for this FTL branch actually exists. If not, then there is no update!
         local path
         path="${ftlBranch}/${binary}"
         # shellcheck disable=SC1090
-        if ! check_download_exists "$path"; then
-            printf "  %b Branch \"%s\" is not available.\\n" "${INFO}" "${ftlBranch}"
-            printf "  %b Use %bpihole checkout ftl [branchname]%b to switch to a valid branch.\\n" "${INFO}" "${COL_LIGHT_GREEN}" "${COL_NC}"
+        check_download_exists "$path"
+        local ret=$?
+        if [ $ret -ne 0 ]; then
+            if [[ $ret -eq 1 ]]; then
+                printf "  %b Branch \"%s\" is not available.\\n" "${INFO}" "${ftlBranch}"
+                printf "  %b Use %bpihole checkout ftl [branchname]%b to switch to a valid branch.\\n" "${INFO}" "${COL_LIGHT_GREEN}" "${COL_NC}"
+                return 2
+            elif [[ $ret -eq 2 ]]; then
+                printf "  %b Unable to download from ftl.pi-hole.net. Please check your Internet connection and try again later.\\n" "${CROSS}"
+                return 3
+            else
+                printf "  %b Unknown error. Please contact Pi-hole Support\\n" "${CROSS}"
+                return 4
+            fi
             return 2
```

Comment:
This line will never be executed.

When the new `if-then-else` command above is executed, one of the `return` commands will be executed before.

_(Suggestion button is not available for this line)_

Reply:
Thanks, fixed